File: leetcode/001_two_sum/two_sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        order = [idx for idx, _ in sorted(enumerate(nums), key=lambda x: x[1])]
        i = 0
        j = len(nums) - 1
        while nums[order[i]] + nums[order[j]] != target and i < j:
            if nums[order[i]] + nums[order[j]] < target:
                i += 1
            else:
                j -= 1
        assert i < j
        return order[i], order[j]

File: leetcode/001_two_sum/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for(int i = 0; i < nums.size(); ++i) {
            int x = target - nums[i];
            auto it = seen.find(x);
            if(it != seen.end()) {
                return {it->second, i};
            }
            seen[nums[i]] = i;
        }
        return {};
    }
};
